### PR TITLE
Update Changelog for v2.44

### PR DESCRIPTION
**Pull request by TAOXUY**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Release 2.44.0 06-04-2023
+
+- Update Envoy to v1.25.4 (#809)
+- Bump API version v11 -> v12 (#808)
+- Revert: Add back in HTTP method override in service control filter (#802) (#806)
```
Comment (nareddyt) [inline CHANGELOG.md:5]: remove, we didn't put original commit in last changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# Release 2.44.0 06-04-2023
+
+- Update Envoy to v1.25.4 (#809)
+- Bump API version v11 -> v12 (#808)
```
Comment (nareddyt) [inline CHANGELOG.md:4]: remove, this is internal API for ESPv2 users

**Review by nareddyt — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (TAOXUY) [reply CHANGELOG.md:4]: We regard CAG as our customers as well and we have changelog for API version update before as well.

https://screenshot.googleplex.com/3jC4ghRxg5aApRu

File: CHANGELOG.md (hunk 1 above)
Comment (TAOXUY) [reply CHANGELOG.md]: Done.

**Pull request merged**